sys-boot/edk2/edk2-2018.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="EFI firmware development environment"
+DESCRIPTION="The EDK II project is the response to the EFI community’s \
+request for a better build and version tracking environment for UEFI and PI \
+development. The main difference between the EDK II to the original EDK is \
+the Enhanced Build Environment of the EDK II. The advantages of the Enhanced \
+Build Environment include:
+
+* Operating System independence
+* Flexibility in choosing the compiler and assembler tools
+* The ability to generate working code using open source build tools and \
+applications
+* Enhanced development and build capability of modules and module packages
+* Use of build configuration tools and data sets to provide flexible process
+* Online source control allows users to contribute code and become participants"
+HOMEPAGE="http://www.tianocore.org/"
+COPYRIGHT="2004-2016 Intel Corporation
+	2008-2010 Apple Inc
+	2011-2015 ARM Limited
+	2014-2015 Linaro Limited
+	2013-2015 Red Hat, Inc."
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/tianocore/edk2/archive/vUDK$portVersion.tar.gz"
+CHECKSUM_SHA256="a37aa6dc7a4326b562ce03e225afd96a0cd1dc847d9aeddfd867ce6b2e7cc7bc"
+SOURCE_DIR="edk2-vUDK$portVersion"
+PATCHES="edk2-$portVersion.patchset"
+DISABLE_SOURCE_PACKAGE="yes"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	edk2 = $portVersion compat >= 0
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libuuid$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:iasl$secondaryArchSuffix
```

Comment:
```diff
-	cmd:iasl$secondaryArchSuffix
+	cmd:iasl$commandSuffix
```

Reply:
What did you meant here?

Reply:
We need to define `commandSuffix` in edk2's recipe because one of its dependencies, `cmd:iasl` is provided by iasl as `cmd:iasl$commandSuffix`.
(I agree, this is a bit annoying, but the only way to avoid this would be to give up using `commandSuffix`.)

Reply:
I still can't get your point. In BUILD_REQ i have cmd:iasl$secondaryArchSuffix

Reply:
If you try to build edk2_x86 on x86_gcc2 it will say it can't find `cmd:iasl_x86`.
That's because the package for iasl provides `cmd:iasl` on x86_gcc2 instead of `cmd:iasl_x86`.

Of course, you could mention `cmd:iasl` instead of `cmd:iasl$secondaryArchSuffix` and it would work on x86_gcc2 but it would not work if you tried to build edk2_x86 on x86_64. So what you need here is exactly `cmd:iasl$commandSuffix` as this is the only expression that is valid in all situations.